src/test/scala/com/cognite/spark/datasource/EventsRelationTest.scala

```diff
@@ -76,6 +76,20 @@ class EventsRelationTest extends FlatSpec with Matchers with SparkTest {
     assert(df.count == 4)
   }
 
+  it should "handle a really advanced query" taggedAs WriteTest in {
+    val df = spark.read
+      .format("com.cognite.spark.datasource")
+      .option("apiKey", writeApiKey.apiKey)
+      .option("type", "events")
+      .load()
+      .where(s"((type = 'maintenance' or type = 'upgrade') " +
+        s"and subtype in('manual', 'automatic')) " +
+        s"or (type = 'maintenance' and subtype = 'manual') " +
+        s"or (type = 'upgrade') and source = 'something'")
+    assert(df.count == 4)
```

Comment:
How would you know how many items that were downloaded to fulfill the query? In other words are we now actually checking if it filters the source system (CDF) itself.

Reply:
Could perhaps check the metrics for this?

Reply:
Good tip! Works perfectly.